src/hyperion/experiment_plans/oav_grid_detection_plan.py

```diff
@@ -150,6 +150,12 @@ def grid_detection_plan(
         yield from bps.abs_set(oav.snapshot.box_width, box_size_x_pixels)
         yield from bps.abs_set(oav.snapshot.num_boxes_x, x_steps)
         yield from bps.abs_set(oav.snapshot.num_boxes_y, y_steps)
+        yield from bps.abs_set(
+            oav.snapshot.microns_per_pixel_x, oav.parameters.micronsPerXPixel
+        )
+        yield from bps.abs_set(
+            oav.snapshot.microns_per_pixel_y, oav.parameters.micronsPerYPixel
+        )
```

Comment:
I think the microns_per_pixel should be on the snapshot as it's a property of the individual snapshot, depending on zoom etc the OAV microns per pixel can change over time, whereas if it's on the snapshot you always know that it relates to the specific image you are looking at. 
Having said that I agree it would be nice if somehow the OAV device itself applied the current mpp to the snapshot at the point at which it is generated.
I didn't do it because at the time it looked a bit fiddly and didn't want to get sidetracked.